docs/userguide/apps.rst

```diff
@@ -112,7 +141,7 @@ Limitations
 There are some limitations on the Python functions that can be converted to apps:
 
 1. Functions should act only on defined input arguments. That is, they should not use script-level or global variables.
-2. Functions must explicitly import any required modules.
+2. Functions must explicitly import any required modules if they are defined in script which starts Parsl.
```

Comment:
i'm fairly certain there's more subtlety here...

Reply:
How so? 

I'm fairly sure it works this way for functions defined in other files. The  main Parsl process will serialize a reference to the location of the function, and the worker will import those libraries as it loads in the function.

That said, my knowledge of Parsl's function serialization is empirical. I don't understand the underlying codebase, just how it has worked for me in practice.

Reply:
pickle does that, but dill does more complex stuff (including changing behaviour depending on where on the filesystem your Other Files are located...) that is hard to keep in my head